parcels/include/parcels.h

```diff
@@ -201,15 +212,10 @@ static inline ErrorCode temporal_interpolation_structured_grid(float x, float y,
 static double dist(float lon1, float lon2, float lat1, float lat2, int sphere_mesh)
 {
   if (sphere_mesh == 1){
-    double r = 360*60*1852 / (2*M_PI);
     double rad = M_PI / 180.;
-    double x1 = r*cos(rad*lon1) * cos(rad*lat1);
-    double y1 = r*sin(rad*lon1) * cos(rad*lat1);
-    double z1 = r*sin(rad*lat1);
-    double x2 = r*cos(rad*lon2) * cos(rad*lat2);
-    double y2 = r*sin(rad*lon2) * cos(rad*lat2);
-    double z2 = r*sin(rad*lat2);
-    return sqrt((x2-x1)*(x2-x1) + (y2-y1)*(y2-y1) + (z2-z1)*(z2-z1));
+    double deg2m = 1852 * 60.;
+    double lat = (lat1+lat2)/2.;
+    return sqrt((lon2-lon1)*(lon2-lon1) * deg2m * deg2m * cos(rad * lat) * cos(rad * lat) + (lat2-lat1)*(lat2-lat1) * deg2m * deg2m);
```

Comment:
This is a different `dist` function than the one in field.py?

Reply:
It shouldn't.
It is the same as ` return np.sqrt(((lon2-lon1)*deg2m*cos(rad * lat))**2 + ((lat2-lat1)*deg2m)**2)`, no?

Reply:
Ah yes, you're right